homeassistant/core.py

```diff
@@ -254,13 +258,15 @@ def async_block_till_done(self):
         """Block till all pending work is done."""
         while True:
             # Wait for the pending tasks are down
-            pending = list(self._pending_tasks)
+            pending = [task for task in self._pending_tasks
+                       if not task.done()]
+            self._pending_tasks.clear()
```

Comment:
I don't think we should clear it. Instead we shouldn't assign to `pending` but to `self._pending_tasks` and just go from there.